pixl_core/tests/conftest.py

```diff
@@ -13,6 +13,157 @@
 #  limitations under the License.
 from __future__ import annotations
 
+import datetime
 import os
+import shutil
+import subprocess
+from pathlib import Path
 
-os.environ["ENV"] = "test"
+import pytest
+from core.database import Base, Extract, Image
+from sqlalchemy import Engine, create_engine
+from sqlalchemy.orm import Session, sessionmaker
+
+os.environ["RABBITMQ_USERNAME"] = "guest"
+os.environ["RABBITMQ_PASSWORD"] = "guest"  # noqa: S105 Hardcoding password
+os.environ["RABBITMQ_HOST"] = "localhost"
+os.environ["RABBITMQ_PORT"] = "25672"
+os.environ["FTP_HOST"] = "localhost"
+os.environ["FTP_USER_NAME"] = "pixl"
+os.environ["FTP_USER_PASS"] = "longpassword"  # noqa: S105 Hardcoding password
+os.environ["FTP_PORT"] = "20021"
+
+TEST_DIR = Path(__file__).parent
+STUDY_DATE = datetime.date.fromisoformat("2023-01-01")
+
+
+@pytest.fixture(scope="package")
+def run_containers() -> None:
+    """Run docker containers for tests which require them."""
+    yield subprocess.run(
+        b"docker compose up --build --wait",
+        check=True,
+        cwd=TEST_DIR,
+        shell=True,  # noqa: S602
+        timeout=60,
+    )
+    subprocess.run(
+        b"docker compose down --volumes",
+        check=True,
+        cwd=TEST_DIR,
+        shell=True,  # noqa: S602
+        timeout=60,
+    )
+
+
+@pytest.fixture(scope="package")
+def data() -> Path:
+    """Directory containing the test data for uploading to the ftp server."""
+    return TEST_DIR / "data"
+
+
+@pytest.fixture(scope="package")
```

Comment:
do we want teardown to happen after each function rather than package, so that if two tests wrote to the same place that we'd be sure that each of them wasn't interacting?

Reply:
Ah yeah, this was due to my repeated attempts to get around file permission errors in Docker volumes on GHA, but ended up "fixing" it by just `ignore_errors=True` when removing files 🙃 
So don't really need this anymore, will fix